### PR TITLE
Feature/109 push generic events

### PR DESCRIPTION
**Pull request by b-brodie-digdir**
Add support for generic events with missing subject field

## Related Issue(s)
- #109 

## Verification
- [x] **Your** code builds clean without any errors or warnings
- [ ] Manual testing done (required)
- [x] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [x] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

File: src/Events/Services/OutboundService.cs [hunk 1]
```diff
@@ -156,7 +160,7 @@ private Uri GetSourceFilter(Uri source)
             }
             else
             {
-                return null;
+                return source;
```
Comment (acn-sbuad) [inline src/Events/Services/OutboundService.cs:163]: not sure why you're changing the return type here? Do we need to update the flow for app events?

File: src/Events/Migration/v0.24/03-setup-function.sql [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    RETURNS TABLE(id bigint, sourcefilter character varying, subjectfilter character varying, typefilter character varying, consumer character varying, endpointurl character varying, createdby character varying, validated boolean, "time" timestamp with time zone) 
+LANGUAGE 'plpgsql'
+AS $BODY$
+
+BEGIN
+return query 
+	SELECT s.id, s.sourcefilter, s.subjectfilter, s.typefilter, s.consumer, s.endpointurl, s.createdby, s.validated, s."time"
+	FROM events.subscription s
+  WHERE  s.sourcefilterhash = ANY(sourcehashset)
+  AND (s.subjectfilter is NULL OR length(subject) = 0 OR s.subjectfilter = subject)
```
Comment (acn-sbuad) [inline src/Events/Migration/v0.24/03-setup-function.sql:14]: do we need to check for null if we're always sending in an empty string from the repository service?

File: src/Events/Services/OutboundService.cs (hunk 1 above)
Comment (b-brodie-digdir) [reply src/Events/Services/OutboundService.cs:163]: GetSourceFilter is used as a cache key. I suggest refactoring to improve readability.

File: src/Events/Migration/v0.24/03-setup-function.sql (hunk 2 above)
Comment (b-brodie-digdir) [reply src/Events/Migration/v0.24/03-setup-function.sql:14]: We don't always send in an empty string as far as I can tell. It should be possible to define a subject on a generic cloud event subscription. Will add a k6 test to verify that the logic is correct.

Comment (b-brodie-digdir) [reply src/Events/Migration/v0.24/03-setup-function.sql]: Reevaluated the insert_subscription codepath and concluded that the existing `events.getsubscriptions` implementation will continue to work now that we always send a non-null value for `subject`. 
Test cases where `subject` is empty string:
subscriptionfilter is null --> return
subscriptionfilter is empty string --> return
subscriptionfilter length > 0 --> do not return

Only thing I don't like is that we don't trim whitespace on string properties in our request models.

**Review by acn-sbuad — APPROVED** [review]
🎈

**Pull request merged**